src/spss/readstat_sav_read.c

```diff
@@ -145,6 +146,187 @@ static readstat_error_t sav_parse_variable_display_parameter_record(sav_ctx_t *c
 static readstat_error_t sav_parse_machine_integer_info_record(const void *data, size_t data_len, sav_ctx_t *ctx);
 static readstat_error_t sav_parse_long_string_value_labels_record(const void *data, size_t size, size_t count, sav_ctx_t *ctx);
 static readstat_error_t sav_parse_long_string_missing_values_record(const void *data, size_t size, size_t count, sav_ctx_t *ctx);
+static readstat_error_t sav_read_multiple_response_sets(size_t data_len, sav_ctx_t *ctx);
+
+static mr_set_t parse_mr_line(const char *line) {
```

Comment:
I reckon this is probably not the best way to design this function (i.e. that it has the return value of the structure we want to obtain). Any guidance on this more than welcome. Should it return the readstat result instead, and update the context that we pass as input argument?